src/node-soap/node-soap-resolver.ts

```diff
@@ -230,6 +231,30 @@ export class NodeSoapWsdlResolver {
         }
     }
 
+    private resolveAnonymousTypeToSoapType(xsdFieldDefinition: XsdFieldDefinition, parentSoapType: SoapObjectType): SoapType {
+        const namespace = xsdFieldDefinition.$targetNamespace;
+        const ownerStringForLog = `field '${xsdFieldDefinition.$name}' of soap type '${parentSoapType.name}'`;
+        this.debug(() => `resolving anonymous type for ${ownerStringForLog} from namespace '${namespace}'`);
+
+        const generatedTypeName = `${parentSoapType.name}_${capitalizeFirstLetter(xsdFieldDefinition.$name)}`;
```

Comment:
Yes, I had that in mind. There is no 100% reliable way to avoid hitting an existing name. Do you think we should add something to make the probability even lower? Could be an `anon` postfix or so.

Reply:
We could do something like this:
```
while (!!this.findXsdTypeDefinition(namespace, generatedTypeName)) {
  generatedTypeName = generatedTypeName + "_"
}
```
Otherwise adding at least a 'Anon' postfix would be good enough for me.

Reply:
It took me a while you do not mean to the word "anon" :D maybe "Anonymous" would be more intuitive to understand

Reply:
I really like @sevenclev 's solution!

Reply:
@henkesn I'll wait on you then before I oublish a new version.